src/StructuredLogViewer.Core/Timeline/Timeline.cs

```diff
@@ -62,8 +40,38 @@ private void Populate(Build build, bool analyzeCpp = false)
                             lane.Add(block);
                         }
                     }
-                });
-            }
+
+                    return;
+                }
+
+                if (node is not TimedNode timedNode)
+                {
+                    return;
+                }
+
+                if (timedNode is Build)
+                {
+                    return;
+                }
+
+                if (timedNode is Microsoft.Build.Logging.StructuredLogger.Task task &&
+                    (string.Equals(task.Name, "MSBuild", StringComparison.OrdinalIgnoreCase) ||
+                     string.Equals(task.Name, "CallTarget", StringComparison.OrdinalIgnoreCase)))
+                {
+                    return;
+                }
+
+                {
+                    var nodeId = timedNode.NodeId;
+                    if (!Lanes.TryGetValue(nodeId, out var lane))
+                    {
+                        lane = new Lane();

```

Comment:
I think there's a race condition here where two threads can both create a lane for the same node id